src/wix/WixToolset.Core.Burn/Bind/BindBundleCommand.cs

```diff
@@ -296,15 +296,13 @@ public void Execute()
                 }
 
                 // Give the embedded payloads without an embedded id yet an embedded id.
-                var payloadIndex = 0;
                 foreach (var payload in payloadSymbols.Values)
                 {
                     Debug.Assert(PackagingType.Unknown != payload.Packaging);
 
                     if (PackagingType.Embedded == payload.Packaging && String.IsNullOrEmpty(payload.EmbeddedId))
                     {
-                        payload.EmbeddedId = String.Format(CultureInfo.InvariantCulture, BurnCommon.BurnAuthoredContainerEmbeddedIdFormat, payloadIndex);
-                        ++payloadIndex;
+                        payload.EmbeddedId = Guid.NewGuid().ToString("N");
```

Comment:
Attached containers are extracted to the same folder with file names taken from the embedded id, later renamed to their correct file names. 
When you have multiple attached containers that all have the same payload ids ('a0', 'a1', ...) these payloads overwrite each other.
The unique id ensures that doesn't happen

Reply:
But the ids were already unique. `payloadIndex` was used across all the containers, it wasn't reset to 0 for each container.

(It's true that dark.exe extracts the attached containers that way. But since it renames the extracted files to the correct names before moving on to the next container, non-unique ids wouldn't have been a problem anyway.)

Reply:
It was 0 for each container in wix3, I didn't check in v4.